### PR TITLE
App isolation

### PR DESCRIPTION
**Pull request by rrati**
Created a new structure to allow multiple apps

### CODE REVIEW

File: cmd/protoform-installer/installer.go [hunk 1]
```diff
@@ -36,22 +37,28 @@ func main() {
...
+	err = installer.Run(stopCh)
+	if err != nil {
+		panic(err)
+	}
 }
 
-func createDefaults() *api.ProtoformDefaults {
-	d := protoform.NewDefaultsObj()
+func createPerceptorAppDefaults() *perceptor.AppConfig {
+	hubPort := 443
```
Comment (msenmurugan) [inline cmd/protoform-installer/installer.go]: Hi @rrati,

Whether you want to use the below function for all int to int32 conversion? This function is inside pkg/hub/common.go

// IntToInt32 will convert from int to int32
func IntToInt32(i int) *int32 {
	j := int32(i)
	return &j
}

File: pkg/apps/perceptor/perceptor.go [hunk 2]
```diff
@@ -28,59 +28,59 @@ import (
...
 }
 
-func (i *Installer) perceptorContainer() *components.Container {
+func (p *App) perceptorContainer() *components.Container {
 	container := components.NewContainer(horizonapi.ContainerConfig{
-		Name:    i.Config.PerceptorImageName,
-		Image:   fmt.Sprintf("%s/%s/%s:%s", i.Config.Registry, i.Config.ImagePath, i.Config.PerceptorImageName, i.Config.PerceptorImageVersion),
+		Name:    p.config.PerceptorImageName,
+		Image:   fmt.Sprintf("%s/%s/%s:%s", p.config.Registry, p.config.ImagePath, p.config.PerceptorImageName, p.config.PerceptorImageVersion),
 		Command: []string{"./perceptor"},
```
Comment (msenmurugan) [inline pkg/apps/perceptor/perceptor.go]: Is that possible to keep have the command derived from image name to support downstream build

File: cmd/protoform-installer/installer.go (hunk 1 above)
Comment (rrati) [reply cmd/protoform-installer/installer.go]: Maybe once that gets merged into the app structure.

File: pkg/apps/perceptor/perceptor.go (hunk 2 above)
Comment (rrati) [reply pkg/apps/perceptor/perceptor.go]: Yes, Will do that for all the components

Comment (msenmurugan) [reply pkg/apps/perceptor/perceptor.go]: you are right Rob except metrics and skyfire

Comment (rrati) [reply pkg/apps/perceptor/perceptor.go]: Actually I think we can do it for skyfire too.  Just not metrics yet.

**Pull request merged**